Bio/PDB/alphafold_db.py

```diff
@@ -0,0 +1,114 @@
+"""A module for interacting with the AlphaFold Protein Structure Database.
+
+See the `database website <https://alphafold.com/>`_ and the `API docs <https://alphafold.com/api-docs/>`_.
+"""
+
+import json
+import os
+from os import PathLike
+
+from Bio.PDB import MMCIFParser
+from Bio.PDB.Structure import Structure as PDBStructure
+from typing import Iterator, Union, Optional
+from urllib.request import urlopen
+
+
+def get_predictions(qualifier: str) -> Iterator[dict]:
+    """Get all AlphaFold predictions for a UniProt accession.
+
+    :param qualifier: A UniProt accession, e.g. P00520
+    :type qualifier: str
+    :return: The AlphaFold predictions
+    :rtype: Iterator[dict]
+    """
+    url = f"https://alphafold.com/api/prediction/{qualifier}"
+    # Retrieve the AlphaFold predictions with urllib
+    with urlopen(url) as response:
+        yield from json.loads(response.read().decode())
+
+
+def _get_mmcif_file_path_for(
+    prediction: dict, directory: Optional[Union[str, bytes, PathLike]] = None
+) -> str:
+    """Get the path to the mmCIF file for an AlphaFold prediction.
+
+    :param prediction: An AlphaFold prediction
+    :type prediction: dict
+    :param directory: The directory that stores the mmCIF file, defaults to the current working directory
+    :type directory: Union[int, str, bytes, PathLike], optional
+    :return: The path to the mmCIF file
+    :rtype: str
+    """
+    if directory is None:
+        directory = os.getcwd()
+
+    cif_url = prediction["cifUrl"]
+    # Get the file name from the URL
+    file_name = cif_url.split("/")[-1]
+    return os.path.join(directory, file_name)
+
+
+def download_cif_for(
+    prediction: dict, directory: Optional[Union[str, bytes, PathLike]] = None
+) -> str:
+    """Download the mmCIF file for an AlphaFold prediction.
+
+    Downloads the file to the current working directory if no destination is specified.
+
+    :param prediction: An AlphaFold prediction
+    :type prediction: dict
+    :param directory: The directory to write the mmCIF data to, defaults to the current working directory
+    :type directory: Union[int, str, bytes, PathLike], optional
+    :return: The path to the mmCIF file
+    :rtype: str
+    """
+    if directory is None:
+        directory = os.getcwd()
+
+    cif_url = prediction["cifUrl"]
+    # Create the directory in case it does not exist
+    os.makedirs(directory, exist_ok=True)
+    file_path = _get_mmcif_file_path_for(prediction, directory)
+
+    if os.path.exists(file_path):
+        print(f"File {file_path} already exists, skipping download.")
+    else:
+        with urlopen(cif_url) as response:
+            data = response.read()
+        # Write the data to destination
+        with open(file_path, "wb") as file:
+            file.write(data)
+
+    return file_path
+
+
+def get_pdb_structures_for(
```

Comment:
Only one nitpick - I'd rename this to `get_structural_models_for` or `get_models_for_uniprot_id` (more explicit). It's just that they're neither PDB formatted files, nor actual structures (scientific nitpicking - a structure is determined from data, a model is predicted).

Reply:
Sounds reasonable to me. I renamed it to `get_structural_models_for`, and I tested it again using the last two manual tests in the description of this PR.